### ISSUE TITLE
i2pd.log spam

### ISSUE THREAD

**Issue by poiuty** [PurpleI2P/i2pd#1822]
```
# du -sh /var/log/i2pd/*
1.2T	/var/log/i2pd/i2pd.log
3.1G	/var/log/i2pd/i2pd.log.2.gz
482M	/var/log/i2pd/i2pd.log.3.gz
28M	/var/log/i2pd/i2pd.log.4.gz
22M	/var/log/i2pd/i2pd.log.5.gz
21M	/var/log/i2pd/i2pd.log.6.gz
```

```
# dd if=/var/log/i2pd/i2pd.log of=/var/log/i2pd/x.log bs=100M count=10
# cat /var/log/i2pd/x.log | tail -n 10
01:04:00@368/error - NTCP2: Accept error Too many open files
01:04:00@368/error - NTCP2: Accept error Too many open files
01:04:00@368/error - NTCP2: Accept error Too many open files
01:04:00@368/error - NTCP2: Accept error Too many open files
01:04:00@368/error - NTCP2: Accept error Too many open files
01:04:00@368/error - NTCP2: Accept error Too many open files
01:04:00@368/error - NTCP2: Accept error Too many open files
01:04:00@368/error - NTCP2: Accept error Too many open files
01:04:00@368/error - NTCP2: Accept error Too many open files
01:04:00@368/error - NTCP2
```

```
# records per second
# cat /var/log/i2pd/x.log | grep "Accept error Too many open files" | sort | uniq -c

  88073 01:02:54@368/error - NTCP2: Accept error Too many open files
 216201 01:02:55@368/error - NTCP2: Accept error Too many open files
  62436 01:02:56@368/error - NTCP2: Accept error Too many open files
 725867 01:02:57@368/error - NTCP2: Accept error Too many open files
 667096 01:02:58@368/error - NTCP2: Accept error Too many open files
 455122 01:02:59@368/error - NTCP2: Accept error Too many open files
 389155 01:03:00@368/error - NTCP2: Accept error Too many open files
  25393 01:03:01@368/error - NTCP2: Accept error Too many open files
 135648 01:03:02@368/error - NTCP2: Accept error Too many open files
 459679 01:03:03@368/error - NTCP2: Accept error Too many open files
 575132 01:03:04@368/error - NTCP2: Accept error Too many open files
 572363 01:03:05@368/error - NTCP2: Accept error Too many open files
 572894 01:03:06@368/error - NTCP2: Accept error Too many open files
 559921 01:03:07@368/error - NTCP2: Accept error Too many open files
 529317 01:03:08@368/error - NTCP2: Accept error Too many open files
 246247 01:03:09@368/error - NTCP2: Accept error Too many open files
 240776 01:03:10@368/error - NTCP2: Accept error Too many open files
 753634 01:03:11@368/error - NTCP2: Accept error Too many open files
 249508 01:03:12@368/error - NTCP2: Accept error Too many open files
 472914 01:03:31@368/error - NTCP2: Accept error Too many open files
 635880 01:03:32@368/error - NTCP2: Accept error Too many open files
 416658 01:03:33@368/error - NTCP2: Accept error Too many open files
 397408 01:03:34@368/error - NTCP2: Accept error Too many open files
 488071 01:03:35@368/error - NTCP2: Accept error Too many open files
 470019 01:03:36@368/error - NTCP2: Accept error Too many open files
 496773 01:03:37@368/error - NTCP2: Accept error Too many open files
 458938 01:03:38@368/error - NTCP2: Accept error Too many open files
 267107 01:03:39@368/error - NTCP2: Accept error Too many open files
 502096 01:03:40@368/error - NTCP2: Accept error Too many open files
 496576 01:03:41@368/error - NTCP2: Accept error Too many open files
 606465 01:03:42@368/error - NTCP2: Accept error Too many open files
 349756 01:03:43@368/error - NTCP2: Accept error Too many open files
 383913 01:03:51@368/error - NTCP2: Accept error Too many open files
 446154 01:03:52@368/error - NTCP2: Accept error Too many open files
 380817 01:03:53@368/error - NTCP2: Accept error Too many open files
 169116 01:03:54@368/error - NTCP2: Accept error Too many open files
 525742 01:03:55@368/error - NTCP2: Accept error Too many open files
 393773 01:03:56@368/error - NTCP2: Accept error Too many open files
 396300 01:03:57@368/error - NTCP2: Accept error Too many open files
 400356 01:03:58@368/error - NTCP2: Accept error Too many open files
 394111 01:03:59@368/error - NTCP2: Accept error Too many open files
 113748 01:04:00@368/error - NTCP2: Accept error Too many open files
```

**Comment by orignal** [comment]
Lack of file descriptions.

Set
ulimit -n 4096 
before start

**Comment by r4sas** [comment]
@poiuty what system are you using and how do you run i2pd?

**Comment by poiuty** [comment]
> Set
> ulimit -n 4096
> before start

Already was.
```
# cat /proc/3509044/limits | grep "Max open files"
Max open files            4096                 4096                 files  
```
Debian 11, change `ulimit` in file `/etc/default/i2pd` dont work. Then try <a href="https://i2pd.readthedocs.io/en/latest/user-guide/run/#overriding-systemd-service-parameters">overriding systemd service parameters</a>.

```
# mkdir -p /etc/systemd/system/i2pd.service.d/
# nano /etc/systemd/system/i2pd.service.d/override.conf

[Service]
LimitNOFILE=16386
```

Check result.
```
# systemctl stop i2pd
# systemctl reenable i2pd
# systemctl daemon-reload
# systemctl start i2pd
# cat /proc/3510249/limits | grep "Max open files"

Max open files            16386                16386                files 
```

**Comment by poiuty** [comment]
It would be nice to fix the spam in log file.
related https://github.com/PurpleI2P/i2pd/issues/1549

**Comment by orignal** [comment]
Yes, I agree it should be fixed.

**Comment by r4sas** [comment]
https://github.com/PurpleI2P/i2pd/commit/737603e81b9b33b1b25fb4a501930dcfffb50016 + https://github.com/PurpleI2P/i2pd/commit/857df5c7341353e8b18af812829797c4da3ccb84

**Comment by reijerh** [comment]
Related to this issue is that I kind of feel that there are still too many messages in the log when I've set the log level to 'error'.

Mostly these:
1. I2CP: Create session duplicate address XXX
2. I2CP: Can't receive header
3. Garlic: Failed to decrypt message
4. Gzip: Deflate error 0
5. Tunnel: Tunnel with id XXX already exists

They seem to be fairly benign and expected as i2pd keeps running just fine, perhaps they should be warnings or debug messages?

They drown out perhaps more errors like `Runtime exception: std::bad_alloc` (this seems bad?).

**Comment by r4sas** [comment]
NTCP2 spamming fixed, try out 2.45.0.
